### PR TITLE
pygmt.grdtrack: Simplify the handling of optional virtual files

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

The `pygmt.grdtrack` method also takes an optional virtual file, so it can be
greatly simplified after https://github.com/GenericMappingTools/pygmt/pull/2493.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
Ping @GenericMappingTools/pygmt-maintainers for reviews.

File: pygmt/src/grdtrack.py
Comment (weiji14) [inline pygmt/src/grdtrack.py:1]: At L286, is it possible to remove `# pylint: disable=too-many-branches`?

Comment (seisman) [reply pygmt/src/grdtrack.py:1]: Removed in 67b8ed7ab1afb006a5008893f55cd693a783b71d.

**Review by weiji14 — APPROVED** [review]
Thanks, looks good now!

Btw, should this be tagged `skip-changelog`?

**Pull request merged**